Testscripts/Linux/verify_vmbus_interrupts.sh

```diff
@@ -45,28 +50,37 @@ function verify_vmbus_interrupts() {
     #
     # Verify if VMBUS interrupts are processed by all CPUs
     #
-    while read line
+    cat /proc/interrupts > interrupts
+    while read -r line
     do
-        if [[ ($line = *hyperv* ) || ( $line = *Hypervisor* ) ]]; then
-            for (( core=0; core<=$cpu_count-1; core++ ))
+        if [[ ($line = *Hyper-V* ) || ( $line = *Hypervisor* ) ]]; then
+            for (( core=0; core<=cpu_count-1; core++ ))
             do
-                intrCount=$(echo "$line" | xargs echo |cut -f $(( $core+2 )) -d ' ')
+                intrCount=$(echo "$line" | xargs echo |cut -f $(( core+2 )) -d ' ')
                 if [ "$intrCount" -ne 0 ]; then
                     (( nonCPU0inter++ ))
-                    UpdateSummary "CPU core ${core} is processing VMBUS interrupts."
+                    UpdateSummary "CPU core $core be processing VMBUS $intrCount interrupts."
                 fi
             done
         fi
-    done < "/proc/interrupts"
+    done < interrupts
 
-    if [ $nonCPU0inter -ne "$cpu_count" ]; then
-        LogErr "Not all CPU cores are processing VMBUS interrupts!"
+    if [ "$nonCPU0inter" -eq 0 ]; then
+        LogMsg "Total CPU counts: $cpu_count"
+        LogMsg "The number of CPU counts using interrupts: $nonCPU0inter"
+        LogErr "None of CPU cores are processing VMBUS interrupts!"
         SetTestStateFailed
-        exit 0
+    elif [ "$nonCPU0inter" -eq "$cpu_count" ]; then
+        LogMsg "All $nonCPU0inter CPU cores are processing interrupts"
+        SetTestStateCompleted
+    elif [ "$nonCPU0inter" -gt "$cpu_count" ]; then
+        LogMsg "All $nonCPU0inter CPU cores are processing interrupts. $((nonCPU0inter - cpu_count)) CPUs are processing multiple interrupts"
+        SetTestStateCompleted
+    else
+        UpdateSummary "$nonCPU0inter CPU cores are processing interrupts."
```

Comment:
when these code will be executed?

Reply:
nonCPU0inter is less than cpu_count but greater than 0.